airflow/jobs/scheduler_job.py

```diff
@@ -798,14 +777,7 @@ def process_file(self, file_path, zombies, pickle_dags=False, session=None):
 
         dags = self._find_dags_to_process(dagbag.dags.values(), paused_dag_ids)
 
-        # Not using multiprocessing.Queue() since it's no longer a separate
-        # process and due to some unusual behavior. (empty() incorrectly
-        # returns true as described in https://bugs.python.org/issue23582 )
-        ti_keys_to_schedule = []
-
-        self._process_dags(dagbag, dags, ti_keys_to_schedule)
-
-        for ti_key in ti_keys_to_schedule:
+        for ti_key in self._process_dags(dagbag, dags, session=session):
```

Comment:
Doesn't this method return TaskInstance now?  We should pass TaskInstance instead of TaskInstance key from what I was checking recently. I wanted to improve it in the near future.  Earlier, we returned the key because we had multiprocessing.

Reply:
Yes, again this was deliberate.

We went from TI, to TIKey, and then back to TI when it was used! This stops an extra DB lookup.

Reply:
Do we want to work on objects like TaskInstance all the time?  _process_dags method return List of TaskInstance, but we assign the result to a variable of type ti_key.  This is confusing.

Reply:
Oh, this was a mistake in my rebase. This should be `for ti in self._process_dags`